cmd/kubeadm/app/util/config/cluster.go

```diff
@@ -111,9 +111,8 @@ func getInitConfigurationFromCluster(kubeconfigDir string, client clientset.Inte
 		return nil, errors.Wrap(err, "failed to get component configs")
 	}
 
-	// if this isn't a new controlplane instance (e.g. in case of kubeadm upgrades)
-	// get nodes specific information as well
-	if !newControlPlane {
+	// in case of kubeadm upgrades get nodes specific information as well
+	if !upgrading {
```

Comment:
I guess that you need to check if `upgrading` is true, and the invert the value in all call to `getInitConfigurationFromCluster`.  And I'm not sure if the `newControlPlane` is false only upgrades cases.

Reply:
yop true thx, i overlooked the  not logic . thx :sun_with_face:

Reply:
thx much @yagonobre :+1: